src/components/sidebar/routePathView/NewRoutePathView.tsx

```diff
@@ -33,24 +37,22 @@ class NewRoutePathView extends React.Component<INewRoutePathViewProps, INewRoute
     constructor(props: any) {
         super(props);
 
-        const routePathStore = this.props.routePathStore;
-        const currentRoutePath = routePathStore ? routePathStore.routePath : null;
-        if (!currentRoutePath) {
-            const newRoutePath = this.createNewRoutePath();
-
-            this.props.routePathStore!.setRoutePath(newRoutePath);
-            this.state = {
-                currentRoutePath: newRoutePath,
-            };
+        const oldRoutePath = this.props.routePathStore!.routePath;
+        let newRoutepath: IRoutePath;
+        if (!oldRoutePath) {
+            newRoutepath = this.createNewRoutePath();
         } else {
-            this.state = {
-                currentRoutePath,
-            };
+            newRoutepath = RoutePathFactory.createNewRoutePathFromOld(oldRoutePath);
```

Comment:
I think this wouldve been enough:
newRoutePath.startTime.setDate(newRoutePath.startTime.getDate() + 1);

But anyway that createNewRoutePathFromOld should be removed when we get timestamp validation working. I suggest to add this: // TODO: remove when timestamp validation works

Reply:
I don't think it would be clear enough what the logic actually means, if we would use the first suggestion. We will probably need to plan how we initiate start times in the future. However, one relatively good solution would to initiate the new routepath, with a suitable starttime, would be to place the logic in the createNewRoutePathFromOld function.

Reply:
Yes, its factory is a right place for creating it if it needs some more complex logic than timestamp + 1. It can be planned in the future